spec/services/action/user/trigger_post_action_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+RSpec.describe Action::User::TriggerPostAction do
```

Comment:
I decided to spec this action in isolation, as it has several branching statements/edge cases. It’s easier like this.